src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -441,6 +477,34 @@ export class CourseManagementService {
         return res;
     }
 
+    /**
+     * Converts the exercise category json string into ExerciseCategory objects (if it exists).
+     * @param res the response
+     * @private
+     */
+    private convertExerciseCategoriesFromServer(res: EntityResponseType): EntityResponseType {
+        if (res.body && res.body.exercises) {
+            res.body.exercises.forEach((exercise) => this.exerciseService.parseExerciseCategories(exercise));
+        }
+        return res;
```

Comment:
We do not return a different object here, so we could change the method return type to void and use `tap` instead of `map`?

Reply:
Actually the object is modified by `parseExerciseCategories`. Should I still use `tap`?

Reply:
Yes, but we're only passing references to objects, so it shouldn't matter if it is modified. I just find it very odd/confusing to return the same object we put in as parameter.